options.js

```diff
@@ -1,139 +1,193 @@
-	chrome.extension.getBackgroundPage().rcxMain.config.popupDelay = popupDelay;
-	chrome.extension.getBackgroundPage().rcxMain.config.disablekeys = localStorage["disablekeys"];
-	chrome.extension.getBackgroundPage().rcxMain.config.showOnKey = localStorage["showOnKey"];
-	chrome.extension.getBackgroundPage().rcxMain.config.kanjicomponents = localStorage["kanjicomponents"];
-	chrome.extension.getBackgroundPage().rcxMain.config.kanjiinfo = kanjiinfoarray;
-	chrome.extension.getBackgroundPage().rcxMain.config.lineEnding = localStorage["lineEnding"];
-	chrome.extension.getBackgroundPage().rcxMain.config.copySeparator = localStorage["copySeparator"];
-	chrome.extension.getBackgroundPage().rcxMain.config.maxClipCopyEntries = localStorage["maxClipCopyEntries"];
-	chrome.extension.getBackgroundPage().rcxMain.config.popupLocation = localStorage["popupLocation"];
-}
-window.onload = fillVals;
-
-/*function clicktab(tab) {
-	selectedtab = document.getElementById(tab);
-	// change format of all tabs to deselected
-	// change format of selected tab to selected
-	// hide all tab contents
-	// show selected tab contents
-}*/
-
-
-document.querySelector('#submit').addEventListener('click', getVals);
-
+
+//Todo: Update config
+//TOdo: add the second tobi addition
+// Get the current options.
+function get_options() {
+	//Todo: Order these
+
+	var getItems = ["popupcolor","highlight","textboxhl","onlyreading",
+		"minihelp","disablekeys", "kanjicomponents","lineEnding", "copySeparator",
+		"maxClipCopyEntries","popupDelay",
+		"showOnKey","popupLocation","kanjiInfo"];
+	chrome.storage.sync.get(getItems,
+		function(items) {
+
+			for(var i=0; i < document.optform.popupcolor.length; ++i) {
+				if(document.optform.popupcolor[i].value == items['popupcolor']) {
+					document.optform.popupcolor[i].selected = true;
+					break;
+				}
+			}
+
+			document.optform.popupLocation.selectedIndex = items.popupLocation;
+			document.optform.highlighttext.checked = items.highlight;
+			document.optform.textboxhl.checked = items.textboxhl;
+			document.optform.onlyreading.checked = items.onlyreading;
+			document.optform.minihelp.checked = items.minihelp;
+			document.optform.disablekeys.checked = items.disablekeys;
+			document.optform.kanjicomponents.checked = items.kanjicomponents;
+			document.optform.popupDelay.value = items.popupDelay;
+
+
+			kanjiInfoLabelList = chrome.extension.getBackgroundPage().rcxDict.prototype.kanjiInfoLabelList;
+
+			for (i = 0; i*2 < kanjiInfoLabelList.length; i++) {
+				// Need to get every other element in the storage, so this funky math was added.
+				// We have abbreviations and values..
+
+				document.getElementById(kanjiInfoLabelList[i*2]).checked = items.kanjiInfo[kanjiInfoLabelList[i*2]] //== 'true' ? true : false;
+			}
+
+	for(var i=0; i < document.optform.lineEnding.length; ++i) {
+		if(document.optform.lineEnding[i].value === items.lineEnding) {
```

Comment:
this for loop and if statement don't seem to be terminated properly.

the save_options function seems to appear in the middle?